src/Application/Console/Contracts/Formatter.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Application\Console\Contracts;
+
+use NunoMaduro\PhpInsights\Domain\Insights\InsightCollection;
+
+/**
+ * This interface is used to define the format of the result.
+ *
+ * @internal
+ */
+interface Formatter
+{
+    /**
+     * Format the result to the desired format.
+     *
+     * @param InsightCollection $insightCollection
+     * @param string $dir
+     * @param array<string> $metrics
```

Comment:
Maybe this should be `@param array<string> $metrics` => `@param array<int, string> $metrics`